File: grade-program.py
# Grade Program
# Wednesday, June 12, 2024 8:00:00 AM EAT
    # Assignment: Grade Program
    # Python

def get_grade(score):
    if score >= 90:
        return "A"
    elif score >= 87:
        return "A (minus)"
    elif score >= 84:
        return "B (plus)"
    elif score >= 80:
        return "B"
    elif score >= 77:
        return "B (minus)"
    elif score >= 74:
        return "C (plus)"
    elif score >= 70:
        return "C"
    elif score >= 67:
        return "C (minus)"
    elif score >= 64:
        return "D (plus)"
    elif score >= 62:
        return "D"
    elif score >= 60:
        return "D (minus)"
    else:
        return "F"

def main():
    while True:
        score = int(input("Enter your score (0-100): "))
        while score < 0 or score > 100:
            print("Invalid score. Please enter a score between 0 and 100.")
            score = int(input("Enter your score (0-100): "))
        
        grade = get_grade(score)
        status = "passed" if grade != "F" else "failed"
        print(f"Your score: {score}, Grade: {grade}, You have {status}.")

        cont = input("Do you want to enter another score? (yes/no): ").strip().lower()
        if cont != "yes":
            break

if __name__ == "__main__":
    main()

File: grade-program.cpp
// Grade Program
// Wednesday, June 12, 2024 8:00:00 AM EAT
    // Assignment: Grade Program
    // C++

#include <iostream>
#include <string>

std::string getGrade(int score) {
    if (score >= 90) return "A";
    else if (score >= 87) return "A (minus)";
    else if (score >= 84) return "B (plus)";
    else if (score >= 80) return "B";
    else if (score >= 77) return "B (minus)";
    else if (score >= 74) return "C (plus)";
    else if (score >= 70) return "C";
    else if (score >= 67) return "C (minus)";
    else if (score >= 64) return "D (plus)";
    else if (score >= 62) return "D";
    else if (score >= 60) return "D (minus)";
    else return "F";
}

int main() {
    int score;
    std::string cont;

    do {
        std::cout << "Enter your score (0-100): ";
        std::cin >> score;

        while (score < 0 || score > 100) {
            std::cout << "Invalid score. Please enter a score between 0 and 100: ";
            std::cin >> score;
        }

        std::string grade = getGrade(score);
        std::string status = (grade != "F") ? "passed" : "failed";

        std::cout << "Your score: " << score << ", Grade: " << grade << ", You have " << status << ".\n";
        std::cout << "Do you want to enter another score? (yes/no): ";
        std::cin >> cont;
    } while (cont == "yes");

    return 0;
}